tests/test_integrated_channels/test_canvas/test_client.py

```diff
@@ -86,3 +89,144 @@ def test_client_instantiation_fails_without_refresh_token(self):
+                transmitter_method(poorly_formatted_data)
+
+        assert client_error.value.__str__() == 'Canvas Client Error: Unable to decode data.'
+
+    def test_course_delete_fails_with_poorly_constructed_data(self):
+        self.update_fails_with_poorly_constructed_data("_delete")
+
+    def test_course_update_fails_with_poorly_constructed_data(self):
+        self.update_fails_with_poorly_constructed_data("_put")
+
+    def update_fails_with_poorly_constructed_data(self, request_type):
+        bad_course_to_update = '{"course": {"name": "test_course"}}'.encode()
+        canvas_api_client = CanvasAPIClient(self.enterprise_config)
+
+        with pytest.raises(CanvasClientError) as client_error:
+            with responses.RequestsMock() as request_mock:
+                request_mock.add(
+                    responses.POST,
+                    self.oauth_url,
+                    json={'access_token': self.access_token},
+                    status=200
+                )
+                transmitter_method = getattr(canvas_api_client, request_type)
+                transmitter_method(bad_course_to_update)
+
+        assert client_error.value.__str__() == 'Canvas Client Error: ' \
+                                               'Could not transmit data, no integration ID present.'
+
+    def test_course_delete_fails_when_course_id_not_found(self):
+        self.update_fails_when_course_id_not_found("_delete")
+
+    def test_course_update_fails_when_course_id_not_found(self):
+        self.update_fails_when_course_id_not_found("_put")
+
+    def update_fails_when_course_id_not_found(self, request_type):
+        course_to_update = '{{"course": {{"integration_id": "{}", "name": "test_course"}}}}'.format(
+            self.integration_id
+        ).encode()
+        mock_all_courses_resp = [
+            {'name': 'wrong course', 'integration_id': 'wrong integration id', 'id': 2}
+        ]
+        canvas_api_client = CanvasAPIClient(self.enterprise_config)
+
+        with pytest.raises(CanvasClientError) as client_error:
+            with responses.RequestsMock() as request_mock:
+                request_mock.add(
+                    responses.GET,
+                    self.get_all_courses_url,
+                    json=mock_all_courses_resp,
+                    status=200
+                )
+                request_mock.add(
+                    responses.POST,
+                    self.oauth_url,
+                    json={'access_token': self.access_token},
+                    status=200
+                )
+                transmitter_method = getattr(canvas_api_client, request_type)
+                transmitter_method(course_to_update)
+
+        assert client_error.value.__str__() == 'Canvas Client Error: ' \
+                                               'No Canvas courses found with associated integration ID.'
+
+    def test_successful_client_update(self):
```

Comment:
I think it would be good to find a way to also call the actual public client level method (non private) in the tests, even though today we are getting full coverage. methods like `create_content_metadata`, `delete_content_metadata` etc (esp if they get changed in the future, then your tests will automatically cover them). just a thought. non blocking.

Reply:
Since as of right now, all the delete and update content metadata methods are just passing the data to the respective delete and put methods, I think this should be possible.

Reply:
The only issue with this and the success case is that the method doesn't return any response or code on success so there's nothing to assert in the test. I've made the change, could you take one last look and make sure you agree with the process?

Reply:
superb!